test/unit/views/movement/ArrivalDetailsViewSpec.scala

```diff
@@ -27,40 +35,154 @@ class ArrivalDetailsViewSpec extends ViewSpec with Injector {
+        emptyView.getElementById("title") must containMessage("arrivalDetails.header")
+      }
+
+      "have date section" which {
+
+        "contains label" in {
+          import scala.collection.JavaConversions._
+
+          emptyView.getElementsByTag("legend").exists { elem =>
+            elem.text() == messages("arrivalDetails.date.question")
+          }
+        }
+
+        "contains hint" in {
+          emptyView.getElementById("dateOfArrival-hint") must containMessage("arrivalDetails.date.hint")
+        }
+
+        "contains input for day" in {
+          emptyView.getElementsByAttributeValue("for", "dateOfArrival_day").first() must containMessage("date.day")
+          emptyView.getElementById("dateOfArrival_day").`val`() mustBe empty
+        }
+
+        "contains input for month" in {
+          emptyView.getElementsByAttributeValue("for", "dateOfArrival_month").first() must containMessage("date.month")
+          emptyView.getElementById("dateOfArrival_month").`val`() mustBe empty
+        }
+
+        "contains input for year" in {
+          emptyView.getElementsByAttributeValue("for", "dateOfArrival_year").first() must containMessage("date.year")
+          emptyView.getElementById("dateOfArrival_year").`val`() mustBe empty
+        }
+      }
+
+      "have time section" which {
+
+        "contains label" in {
+          import scala.collection.JavaConversions._
+
+          emptyView.getElementsByTag("legend").exists { elem =>
+            elem.text() == messages("arrivalDetails.time.question")
+          }
+        }
+
+        "contains hint" in {
+          emptyView.getElementById("timeOfArrival-hint") must containMessage("arrivalDetails.time.hint")
+        }
+
+        "contains input for hour" in {
+          emptyView.getElementsByAttributeValue("for", "timeOfArrival_hour").first() must containMessage("time.hour")
+          emptyView.getElementById("timeOfArrival_hour").`val`() mustBe empty
+        }
+
+        "contains input for minute" in {
+          emptyView.getElementsByAttributeValue("for", "timeOfArrival_minute").first() must containMessage("time.minute")
+          emptyView.getElementById("timeOfArrival_minute").`val`() mustBe empty
+        }
+      }
+
+      "have 'Continue' button" in {
+        emptyView.getElementsByClass("govuk-button").first() must containMessage("site.continue")
+      }
     }
 
-    "render heading input with hint for time" in {
-      page(movementDetails.arrivalForm(), None).getElementById("timeOfArrival-hint") must containMessage("arrivalDetails.time.hint")
+    "provided with form containing data" should {
+      val date = LocalDate.now().minusDays(1)
+      val time = LocalTime.of(1, 2)
+      val viewWithData = createView(movementDetails.arrivalForm().fill(ArrivalDetails(Date(date), Time(time))))
+
+      "have value in day field" in {
+        viewWithData.getElementById("dateOfArrival_day").`val`() mustBe convertIntoTwoDigitFormat(date.getDayOfMonth)
```

Comment:
Why didn't you just hard-code the expected 2/4 digit values?  I can't see any tests for the two `convert` methods.

Reply:
These methods are very simple and located at the beginning of the file. The question is not to me actually, because I just copied these into a proper place.
And you cannot hard-code the values, because the validation is date-dependent, so the input data is generated new every execution.

Reply:
You could create the test dates with known values rather than relying on "now".    I know it won't make a difference in this case, but tests using current date/time can be a recipe for failing at some point in the future (usually when transitioning into or out of daylight savings).

Anyway, no change required.  Happy to approve once the 'change' links are re-inrtoduced.